src/components/[guild]/Events/EventCard.tsx

```diff
@@ -0,0 +1,40 @@
+import {
+  Modal,
+  ModalCloseButton,
+  ModalContent,
+  ModalOverlay,
+} from "@chakra-ui/react"
+import { PropsWithChildren, useState } from "react"
+
+type Props = {
+  modal: JSX.Element
+}
+
+const EventCard = ({ children, modal }: PropsWithChildren<Props>): JSX.Element => {
+  const [isOpen, showModal] = useState(false)
+
+  return (
+    <>
+      <div
+        onClick={() => {
+          showModal(true)
+        }}
+        style={{ width: "100%" }}
+      >
+        {children}
+      </div>
+      <Modal
```

Comment:
This is not that much extra code, so I'd just put it inside `DiscordEventCard`

Reply:
I would keep this component as a wrapper for the upcoming events.